### PR TITLE
Bugfix: CLS should clear current active buffer

### LINKED ISSUE #1189: CLS doesn't work in the VT alternate buffer
# Environment

```none
Windows build number: [Version 10.0.18362.145]

Windows Terminal version (if applicable): n/a

Any other software? wsl, nano
```

# Steps to reproduce
1) In either powershell or cmd in terminal, type `wsl nano`
2) Execute `taskkill -im wsl.exe -f` to kill wsl
3) Type `cls` in the terminal where wsl was open

# Expected behavior
`cls` should clear the screen and make the terminal behave like it used to before `wsl nano`

# Actual behavior
after killing `wsl` powershell/cmd is in a bugged state because nano takes up the whole screen
cmd: cls won't clear the screen (first screenshot)
powershell: cls will clear screen BUT if the user inputs e.g. "cls", terminal will print "cclcls", yet still interpret it as "cls" (second screenshot)

L0030755

L0030807

This issue is also present in the original cmd/powershell.

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
CLS calls two functions: 
- `SetConsoleCursorPositionImpl()`
- `ScrollConsoleScreenBufferWImpl()`
Both of these were not checking which buffer to apply to (main vs active buffer).

Now we get the active buffer and apply the changes to that one.

Also, we forgot to switch out of the alt buffer in the previous test. Added that in.

## PR Checklist
* [x] Closes #1189 
* [x] CLA signed.
* [x] Tests added/passed
* [x] ~Requires documentation to be updated~
* [x] I am a core contributor.

## Validation Steps Performed
manually performed what is described in the attached bug report.

Also ran automated test.

### CODE REVIEW

File: src/host/getset.cpp [hunk 1]
```diff
@@ -835,18 +839,20 @@ void ApiRoutines::GetLargestConsoleWindowSizeImpl(const SCREEN_INFORMATION& cont
         //      this scenario is highly unlikely, and we can reasonably do this
         //      on their behalf.
         // see MSFT:19853701
+
+        // use context (not buffer) here because alternate buffer may not be aware of VT mode
```
Comment (DHowett-MSFT) [inline src/host/getset.cpp]: So chatting with @zadjii-msft made me more confident that it's fine to use `buffer` here. It's even _more_ possible that this code won't even be necessary with ScrollRegion being as okay as it is

File: src/host/getset.cpp [hunk 2]
```diff
@@ -835,18 +839,20 @@ void ApiRoutines::GetLargestConsoleWindowSizeImpl(const SCREEN_INFORMATION& cont
         //      this scenario is highly unlikely, and we can reasonably do this
         //      on their behalf.
         // see MSFT:19853701
+
+        // use context (not buffer) here because alternate buffer may not be aware of VT mode
         if (context.InVTMode())
```
Comment (zadjii-msft) [inline src/host/getset.cpp]:
```suggestion
        if (buffer.InVTMode())
```

Comment (zadjii-msft) [reply src/host/getset.cpp]: Okay the spacing looks weird in the GH preview, just change this locally to be safe

File: src/host/getset.cpp (hunk 1 above)
Comment (DHowett-MSFT) [reply src/host/getset.cpp]: Oh gods we investigated this
It's true, it is correct, the alt buffer has the same modes as the main buffer. It is propagated incidentally through a magic global. :smile:

Comment (DHowett-MSFT) [reply src/host/getset.cpp]: Incidentally and in a way that could make it look like an accident.

**Pull request merged**